### DISCUSSION TITLE
Why does the recording quality differs from android sample ?

### DISCUSSION THREAD

**Discussion by florianmlr** [pedroSG94/RootEncoder#985] [General]
Hello,
As I'm looking to improve the quality of the recording, I've made several quality comparison on different repository by using the same phone (pixel 4a), i've noticed the quality of recording on theses sample : https://github.com/android/camera-samples/tree/main/Camera2Video https://github.com/android/camera-samples/tree/main/CameraXVideo are better than the samples on this project ( like the RtmpActivity) , all sample configured with 1280*720 , also the size of the recorded videos in the android sample are 4 times bigger than in the sample of this project and with the same video length. 
As far i know they are using the same video encoder H264
I probably missed some parameters in my test and I would like to understand what makes such big differences.

**Comment by pedroSG94** [comment]
Hello,

The video resolution is not all for video quality.
This library is oriented to streaming for this reason the video bitrate is reduced to reduce bandwidth in my examples but you can configure it as you want.
If you want record with the same quality you can increase video bitrate.
For example:
https://github.com/android/camera-samples/blob/main/Camera2Video/app/src/main/java/com/example/android/camera2/video/fragments/CameraFragment.kt#L405
In this case video bitrate is 10 times higher than my example (1_200_000)

Try use exactly same paramaters (iframe interval, resolution, fps and bitrate) and you will get a similar result and, of course, similar size.

**Comment by florianmlr** [comment]
Thank you for this fast reply and well explanation. You're right with such bitrate I've observed the same quality, however I have noticed some frame drop in the video, but that's not so important.
Do you know if there is a recommended bitrate for a particular resolution ? Does the bitrate should be determined according to the bandwidth and is there a nice way to determine it instead of hardcoding it in android ?

**Reply by pedroSG94** [reply]
About recommended values you can try check plaforms recomendations. For example YouTube.
This is for stream:
https://support.google.com/youtube/answer/2853702?hl=en#zippy=
And this is for video:
https://support.google.com/youtube/answer/1722171?hl=en#zippy=%2Cbitrate
Also, you can change video bitrate on fly while you are streaming or recording:
https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/wiki/Adaptative-video-bitrate
